### PR TITLE
feat: Experiments improvements

### PR DESCRIPTION
**Pull request by anticorrelator**
splitting up #3616

### CODE REVIEW

**Review by mikeldking — APPROVED** [review]

File: src/phoenix/datasets/evaluators/code_evaluators.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+class MatchesRegex:
+    annotator_kind = "CODE"
+
+    def __init__(self, pattern: Union[str, re.Pattern[str]], name: Optional[str] = None) -> None:
+        if isinstance(pattern, str):
+            pattern = re.compile(pattern)
+        self.pattern = pattern
+        assert isinstance(pattern, re.Pattern)
+        self.name = name or f"MatchesRegex({pattern})"
```
Comment (mikeldking) [inline src/phoenix/datasets/evaluators/code_evaluators.py:102]:
```suggestion
        self.name = name or f"matches_({pattern})"
```

File: src/phoenix/datasets/experiments.py [hunk 2]
```diff
@@ -458,6 +458,7 @@ async def async_evaluate_run(
         max_retries=0,
         exit_on_error=False,
         fallback_return_value=None,
+        tqdm_bar_format=get_tqdm_progress_bar_formatter("running experiment evaluations"),
```
Comment (mikeldking) [inline src/phoenix/datasets/experiments.py:461]: @anticorrelator can this be the eval name?

Comment (anticorrelator) [reply src/phoenix/datasets/experiments.py:461]: yeah this is higher level, we're grouping all the evals together and they're being done in parallel now

Comment (mikeldking) [reply src/phoenix/datasets/experiments.py:461]: hmmm, okay gotcha... I sorta want a progress bar per but can wait

**Pull request merged**